dash/_callback_context.py

```diff
@@ -247,5 +248,20 @@ def using_outputs_grouping(self):
     def timing_information(self):
         return getattr(flask.g, "timing_information", {})
 
+    @has_context
+    def set_props(self, component_id: typing.Union[str, dict], props: dict):
+        ctx_value = _get_context_value()
+        if isinstance(component_id, dict):
+            ctx_value.updated_props[json.dumps(component_id)] = props
```

Comment:
Does plain `json.dumps` work here? It's not quite the same as we have for stringifying component IDs in the callback definition:

https://github.com/plotly/dash/blob/9a4a47933e541ad0a61fd377314b576f93a2ea37/dash/dependencies.py#L46-L59

For `set_props` we don't need to handle wildcards, but I'd have thought we still want to sort keys and compress spaces.
(though looks like `_dump` there is unused and should be removed)

Reply:
There is already a `stringify_id` here: https://github.com/plotly/dash/blob/9a4a47933e541ad0a61fd377314b576f93a2ea37/dash/_utils.py#L169-L172

I think the output is the same and is used in the base component, we should use the same everywhere.